File: 0217-contains-duplicate/0217-contains-duplicate.go
func containsDuplicate(nums []int) bool {
    i := 0
    counts := make(map[int]int)
    for i < len(nums) {
        counts[nums[i]]++;
        if counts[nums[i]] > 1 {
            return true;
        }
        i++;
    }
    return false;
}

File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    const bool containsDuplicate(const vector<int>& nums) {
        unordered_set<int> numSet;
        for (const int &n : nums) {
            if (numSet.contains(n)) return true;
            numSet.insert(n);
        }
        return false;
    }
};